### PR TITLE
[ADD] stock_picking_service_sale: new module

### PR DESCRIPTION
**Pull request by agaldona**
TO-DO:

- [x] Readme
- [x] Tests
- [x] ACL

### CODE REVIEW

File: stock_picking_service_sale/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    sale_service_lines = fields.One2many(
+        comodel_name='stock.sale.service.line', inverse_name='picking_id',
+        string='Sale Service Lines')
+    amount_untaxed = fields.Float(compute='_amount_all')
+    amount_tax = fields.Float(compute='_amount_all')
+    amount_total = fields.Float(compute='_amount_all')
+
+    @api.model
+    def _create_backorder(self, picking, backorder_moves=[]):
```
Comment (oihane) [inline stock_picking_service_sale/models/stock_picking.py]:
```python
def _create_backorder(self, picking, backorder_moves=None):
    if not backorder_moves:
        backorder_moves = []
```

Para pasar el error del pylint

**Review by Daniel-CA — APPROVED** [review]

**Comment by agaldona** [comment]
New module to charge services in pickings.
 - New service line object, each line has a direct link to sale line,
   shows sale line's unit price, quantity, discount and subtotal fields and
   a check which when is deactivated moves the service line to the backorder picking.
 - When a picking is created from a sale order, the system will create one service
   line related to this picking for each line in the sale order with service product.
 - When transfering a partial picking, all the lines with deactivated check will be
   moved to the backorder picking. And all lines with activated check will stay in
   the transferred picking.
 - In picking valuation will sum all service charges.
 - When invoicing a picking, the system will not invoice all service lines in the
   sale order, only the lines related to the picking.
 - The service lines will be displayed in the picking report.

@avanzosc/developers Review when possible please

File: stock_picking_service_sale/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_ship_create(self):
+        res = super(SaleOrder, self).action_ship_create()
+        picking = self.env['stock.picking'].search(
+            [('group_id', '=', self.procurement_group_id.id)])
+        picking.get_service_lines(self)
```
Comment (alfredoavanzosc) [inline stock_picking_service_sale/models/sale_order.py:17]: ¿Siempre va a encontrar un albarán?

File: stock_picking_service_sale/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    amount_untaxed = fields.Float(compute='_amount_all')
+    amount_tax = fields.Float(compute='_amount_all')
+    amount_total = fields.Float(compute='_amount_all')
+
+    @api.model
+    def _create_backorder(self, picking, backorder_moves=None):
+        if not backorder_moves:
+            backorder_moves = []
+        res = super(StockPicking, self)._create_backorder(
+            picking, backorder_moves=backorder_moves)
```
Comment (alfredoavanzosc) [inline stock_picking_service_sale/models/stock_picking.py]: def _create_backorder(self, picking, backorder_moves=None):

        res = super(StockPicking, self)._create_backorder(
                 picking, backorder_moves=backorder_moves or [])

File: stock_picking_service_sale/models/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    @api.multi
+    @api.depends('move_lines', 'move_lines.product_qty',
+                 'move_lines.product_uos_qty', 'sale_service_lines')
+    def _amount_all(self):
+        for picking in self:
+            picking.amount_untaxed = 0.0
+            picking.amount_total = 0.0
+            val2 = val1 = val = 0.0
+            for line in picking.move_lines:
+                if line.procurement_id and line.procurement_id.sale_line_id:
```
Comment (alfredoavanzosc) [inline stock_picking_service_sale/models/stock_picking.py]: Esto creo que se puede poner en una sola condición:
if line.procurement_id.sale_line_id:

**Comment by anajuaristi** [comment]
@Daniel-CA porfa, revisa la última versión y si está OK, lo mergeais ya. 
Hay que subirlo al servidor de CMSA

**Pull request merged**